tracing/scope.go

```diff
@@ -249,6 +356,155 @@ func NamedScopeError(ctx context.Context, spanName string, action ScopeAction, o
 	return err
 }
 
+// CallScopeBranch calls action function within a tracing span named after the
+// calling function.
+// Equivalent to wrapping a code block with `BranchScope()`/`EndScope()`.
+func CallScopeBranch(ctx context.Context, action ScopeAction, opts ...trace.SpanStartOption) error {
+	if !trace.SpanContextFromContext(ctx).IsValid() {
+		return action(ctx)
+	}
+	level := InfoLevel
+	spanName, fileTag := getCallerSpanName(2)
+	ctx = startSpan(ctx, spanName, fileTag, level, opts...)
+	err := action(ctx)
+	EndScope(ctx, err)
+	return err
+}
+
+// CallScopeBranchDebug calls action function within a tracing span named after the
+// calling function.  Scope tagged with log level debug.
+// Equivalent to wrapping a code block with `BranchScopeDebug()`/`EndScope()`.
+func CallScopeBranchDebug(ctx context.Context, action ScopeAction, opts ...trace.SpanStartOption) error {
+	if !trace.SpanContextFromContext(ctx).IsValid() {
+		return action(ctx)
+	}
+	level := DebugLevel
+	spanName, fileTag := getCallerSpanName(2)
+	ctx = startSpan(ctx, spanName, fileTag, level, opts...)
+	err := action(ctx)
+	EndScope(ctx, err)
+	return err
+}
+
+// CallScopeBranchInfo calls action function within a tracing span named after the
+// calling function.  Scope tagged with log level info.
+// Equivalent to wrapping a code block with `BranchScopeInfo()`/`EndScope()`.
+func CallScopeBranchInfo(ctx context.Context, action ScopeAction, opts ...trace.SpanStartOption) error {
+	if !trace.SpanContextFromContext(ctx).IsValid() {
+		return action(ctx)
+	}
+	level := InfoLevel
+	spanName, fileTag := getCallerSpanName(2)
+	ctx = startSpan(ctx, spanName, fileTag, level, opts...)
+	err := action(ctx)
+	EndScope(ctx, err)
+	return err
+}
+
+// CallScopeBranchWarn calls action function within a tracing span named after the
+// calling function.  Scope tagged with log level warn.
+// Equivalent to wrapping a code block with `BranchScopeWarn()`/`EndScope()`.
+func CallScopeBranchWarn(ctx context.Context, action ScopeAction, opts ...trace.SpanStartOption) error {
+	if !trace.SpanContextFromContext(ctx).IsValid() {
+		return action(ctx)
+	}
+	level := WarnLevel
+	spanName, fileTag := getCallerSpanName(2)
+	ctx = startSpan(ctx, spanName, fileTag, level, opts...)
+	err := action(ctx)
+	EndScope(ctx, err)
+	return err
+}
+
+// CallScopeBranchError calls action function within a tracing span named after the
+// calling function.  Scope tagged with log level error.
+// Equivalent to wrapping a code block with `BranchScopeError()`/`EndScope()`.
+func CallScopeBranchError(ctx context.Context, action ScopeAction, opts ...trace.SpanStartOption) error {
+	if !trace.SpanContextFromContext(ctx).IsValid() {
+		return action(ctx)
+	}
+	level := ErrorLevel
```

Comment:
Could it be a const?

Reply:
I ended up removing the `level` var entirely.